src/cloud-element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -1032,6 +1034,138 @@ export default class ChangeElementTemplateHandler {
 
     return replacedElement;
   }
+
+
+  _updateLinkedResources(element, oldTemplate, newTemplate) {
+    const bpmnFactory = this._bpmnFactory,
+          commandStack = this._commandStack;
+
+    const newLinkedResources = newTemplate.properties.filter((newProperty) => {
+      const newBinding = newProperty.binding,
+            newBindingType = newBinding.type;
+
+      return newBindingType === 'zeebe:linkedResource';
+    });
+
+    const extensionElements = this._getOrCreateExtensionElements(element);
+
+    let linkedResources = findExtension(extensionElements, 'zeebe:LinkedResources');
+
+    // (1) remove linkedResourcess if no new specified
+    if (!newLinkedResources.length) {
+      if (!linkedResources) {
+        return;
+      }
+
+      commandStack.execute('element.updateModdleProperties', {
+        element,
+        moddleElement: extensionElements,
+        properties: {
+          values: without(extensionElements.get('values'), linkedResources)
+        }
+      });
+      return;
+    }
+
+    if (!linkedResources) {
+      linkedResources = bpmnFactory.create('zeebe:LinkedResources');
+
+      linkedResources.$parent = extensionElements;
+
+      commandStack.execute('element.updateModdleProperties', {
+        element,
+        moddleElement: extensionElements,
+        properties: {
+          values: [ ...extensionElements.get('values'), linkedResources ]
+        }
+      });
+    }
+
+    const unusedLinkedResources = linkedResources.get('values')?.slice() || [];
```

Comment:
An array property in moddle returns always an array. No need for optional chaining IMO.

```suggestion
    const unusedLinkedResources = linkedResources.get('values').slice();
```